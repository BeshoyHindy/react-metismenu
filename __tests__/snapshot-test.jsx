/**
 * __tests__/menu-test.js
 * Author: H.Alper Tuna <halpertuna@gmail.com>
 * Date: 20.08.2016
 */

/* eslint-env jest */
/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */

import React from 'react';
import renderer from 'react-test-renderer';

import Menu from '../src/MetisMenu';

describe('Menu item has submenu', () => {
  // Create test component
  const content = [
    { label: 'Item 1', href: '#item-1', icon: 'bolt' },
    {
      label: 'Item 2',
      href: '#item-2',
      icon: 'bell',
      content: [
        { label: 'Sub Item 1', href: '#sub-item-1', icon: 'eye' },
        { label: 'Sub Item 2', href: '#sub-item-2', icon: 'globe' },
      ],
    },
    {
      label: 'Item 3',
      href: '#item-3',
      icon: 'bell',
      content: [
        { label: 'Sub Item 3', href: '#sub-item-3', icon: 'eye' },
        {
          label: 'Sub Item 4',
          href: '#sub-item-4',
          icon: 'globe',
          content: [
            { label: 'Level 3 Item 1', href: '#level-3-item-1', icon: 'bolt' },
            { label: 'Level 3 Item 2', href: '#level-3-item-2', icon: 'bell' },
          ],
        },
      ],
    },
  ];
  const fakeEvent = {
    preventDefault() {},
  };
  const component = renderer.create(<Menu content={content} />);
  let tree = component.toJSON();

  it('looks like', () => {
    // First snapshot
    expect(tree).toMatchSnapshot();
  });

  it('changes class and caret icon when clicked', () => {
    // Manually click third link (tree > ul:0 > li:2 > a:0, "Item 3")
    tree.children[0].children[2].children[0].props.onClick(fakeEvent);
    // Re-render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // Manually click third link's sub menu (tree > ul:0 > li:2 > ul:1 > li:1 > a:0, "Sub Item 4")
    tree.children[0].children[2].children[1].children[1].children[0].props.onClick(fakeEvent);
    // Re-render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('collapses peer items and submenus of them when clicked', () => {
    // Manually click second link (tree > ul:0 > li:1 > a:0, "Item 2")
    tree.children[0].children[1].children[0].props.onClick(fakeEvent);
    // Re-render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
