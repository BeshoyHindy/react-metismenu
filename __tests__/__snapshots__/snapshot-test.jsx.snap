exports[`Menu item has submenu changes class and caret icon when clicked 1`] = `
<div
  className="metismenu">
  <ul
    className="metismenu-container">
    <li
      className="metismenu-item">
      <a
        href="#item-1"
        onClick={false}
        target="">
        <span
          className="metismenu-icon  fa fa-bolt" />
        Item 1
      </a>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 2
        <span
          className="metismenu-iconlevel fa fa-caret-left" />
      </a>
      <ul
        className="metismenu-container">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-1"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 1
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#sub-item-2"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 2
          </a>
        </li>
      </ul>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 3
        <span
          className="metismenu-iconlevel fa fa-caret-down" />
      </a>
      <ul
        className="metismenu-container visible">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-3"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 3
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#"
            onClick={[Function bound toggleContainer]}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 4
            <span
              className="metismenu-iconlevel fa fa-caret-left" />
          </a>
          <ul
            className="metismenu-container">
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-1"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bolt" />
                Level 3 Item 1
              </a>
            </li>
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-2"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bell" />
                Level 3 Item 2
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Menu item has submenu changes class and caret icon when clicked 2`] = `
<div
  className="metismenu">
  <ul
    className="metismenu-container">
    <li
      className="metismenu-item">
      <a
        href="#item-1"
        onClick={false}
        target="">
        <span
          className="metismenu-icon  fa fa-bolt" />
        Item 1
      </a>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 2
        <span
          className="metismenu-iconlevel fa fa-caret-left" />
      </a>
      <ul
        className="metismenu-container">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-1"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 1
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#sub-item-2"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 2
          </a>
        </li>
      </ul>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 3
        <span
          className="metismenu-iconlevel fa fa-caret-down" />
      </a>
      <ul
        className="metismenu-container visible">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-3"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 3
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#"
            onClick={[Function bound toggleContainer]}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 4
            <span
              className="metismenu-iconlevel fa fa-caret-down" />
          </a>
          <ul
            className="metismenu-container visible">
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-1"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bolt" />
                Level 3 Item 1
              </a>
            </li>
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-2"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bell" />
                Level 3 Item 2
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Menu item has submenu collapses peer items and submenus of them when clicked 1`] = `
<div
  className="metismenu">
  <ul
    className="metismenu-container">
    <li
      className="metismenu-item">
      <a
        href="#item-1"
        onClick={false}
        target="">
        <span
          className="metismenu-icon  fa fa-bolt" />
        Item 1
      </a>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 2
        <span
          className="metismenu-iconlevel fa fa-caret-down" />
      </a>
      <ul
        className="metismenu-container visible">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-1"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 1
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#sub-item-2"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 2
          </a>
        </li>
      </ul>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 3
        <span
          className="metismenu-iconlevel fa fa-caret-left" />
      </a>
      <ul
        className="metismenu-container">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-3"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 3
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#"
            onClick={[Function bound toggleContainer]}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 4
            <span
              className="metismenu-iconlevel fa fa-caret-left" />
          </a>
          <ul
            className="metismenu-container">
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-1"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bolt" />
                Level 3 Item 1
              </a>
            </li>
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-2"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bell" />
                Level 3 Item 2
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`Menu item has submenu looks like 1`] = `
<div
  className="metismenu">
  <ul
    className="metismenu-container">
    <li
      className="metismenu-item">
      <a
        href="#item-1"
        onClick={false}
        target="">
        <span
          className="metismenu-icon  fa fa-bolt" />
        Item 1
      </a>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 2
        <span
          className="metismenu-iconlevel fa fa-caret-left" />
      </a>
      <ul
        className="metismenu-container">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-1"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 1
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#sub-item-2"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 2
          </a>
        </li>
      </ul>
    </li>
    <li
      className="metismenu-item">
      <a
        href="#"
        onClick={[Function bound toggleContainer]}
        target="">
        <span
          className="metismenu-icon  fa fa-bell" />
        Item 3
        <span
          className="metismenu-iconlevel fa fa-caret-left" />
      </a>
      <ul
        className="metismenu-container">
        <li
          className="metismenu-item">
          <a
            href="#sub-item-3"
            onClick={false}
            target="">
            <span
              className="metismenu-icon  fa fa-eye" />
            Sub Item 3
          </a>
        </li>
        <li
          className="metismenu-item">
          <a
            href="#"
            onClick={[Function bound toggleContainer]}
            target="">
            <span
              className="metismenu-icon  fa fa-globe" />
            Sub Item 4
            <span
              className="metismenu-iconlevel fa fa-caret-left" />
          </a>
          <ul
            className="metismenu-container">
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-1"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bolt" />
                Level 3 Item 1
              </a>
            </li>
            <li
              className="metismenu-item">
              <a
                href="#level-3-item-2"
                onClick={false}
                target="">
                <span
                  className="metismenu-icon  fa fa-bell" />
                Level 3 Item 2
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
